nilearn/decomposition/base.py

```diff
@@ -0,0 +1,533 @@
+"""
+Base class for decomposition estimators, utilies for masking and reducing group
+data
+"""
+from __future__ import division
+
+import atexit
+import os
+from math import ceil
+from tempfile import mkstemp, mkdtemp
+import warnings
+import numpy as np
+from scipy import linalg
+from sklearn.externals.joblib import Memory, Parallel, delayed
+from sklearn.linear_model import LinearRegression
+from sklearn.utils import check_random_state
+from sklearn.utils.extmath import randomized_svd
+from sklearn.base import BaseEstimator
+from ..input_data import NiftiMapsMasker
+from ..input_data.masker_validation import check_embedded_nifti_masker
+from .._utils.cache_mixin import CacheMixin, cache
+from .._utils.niimg_conversions import check_niimg_4d
+from .._utils.niimg import _safe_get_data
+
+class MaskReducer(BaseEstimator):
+    """Mask and reduce provided data with provided masker, using a PCA
+
+    Uses a PCA (randomized for small reduction ratio) or a range finding matrix
+    on time series to reduce data size in time. For multiple image,
+    the concatenation of data is returned, either as an ndarray or a memorymap
+    (useful for big datasets that do not fit in memory).
+
+    Parameters
+    ----------
+    masker: NiftiMasker or MultiNiftiMasker
+        Masker to use to mask provided data
+
+    imgs: list of Niimg-like objects
+        See http://nilearn.github.io/building_blocks/manipulating_mr_images.html#niimg.
+        List of subject data
+
+    confounds: CSV file path or 2D matrix
+        This parameter is passed to signal.clean. Please see the
+        corresponding documentation for details.
+
+    reduction_ratio: 'auto' or float
```

Comment:
This explanation does not help much in understanding what this argument actually does.